grammars/php.cson

```diff
@@ -487,233 +480,506 @@
               {
-                'include': 'source.json'
+                'match': '\\\\[\\\\\'\\[\\]]'
+                'name': 'constant.character.escape.php'
               }
             ]
           }
           {
-            'begin': '(<<<)\\s*([\'"]?)(CSS)(\\2)\\s*$\\n?'
+            'match': '[$^+*]'
+            'name': 'keyword.operator.regexp.php'
+          }
+          {
+            'begin': '(?<=^|\\s)(#)\\s(?=[[a-zA-Z0-9,. \\t?!-][^\\x{00}-\\x{7F}]]*$)'
             'beginCaptures':
-              '0':
-                'name': 'punctuation.section.embedded.begin.php'
               '1':
-                'name': 'punctuation.definition.string.php'
-              '3':
-                'name': 'keyword.operator.heredoc.php'
-            'contentName': 'source.css'
-            'end': '^(\\3)\\b'
+                'name': 'punctuation.definition.comment.php'
+            'comment': 'We are restrictive in what we allow to go after the comment character to avoid false positives, since the availability of comments depend on regexp flags.'
+            'end': '$\\n?'
             'endCaptures':
               '0':
-                'name': 'punctuation.section.embedded.end.php'
-              '1':
-                'name': 'keyword.operator.heredoc.php'
-            'name': 'meta.embedded.css'
-            'patterns': [
-              {
-                'include': 'source.css'
-              }
-            ]
+                'name': 'punctuation.definition.comment.php'
+            'name': 'comment.line.number-sign.php'
           }
+        ]
+      }
+      {
+        'begin': '(<<<)\\s*("?)([a-zA-Z_]+[a-zA-Z0-9_]*)(\\2)(\\s+)?'
+        'beginCaptures':
+          '1':
+            'name': 'punctuation.definition.string.php'
+          '3':
+            'name': 'keyword.operator.heredoc.php'
+          '5':
+            'name': 'invalid.illegal.trailing-whitespace.php'
+        'end': '^(\\3)\\b'
+        'endCaptures':
+          '1':
+            'name': 'keyword.operator.heredoc.php'
+        'patterns': [
           {
-            'begin': '(<<<)\\s*([\'"]?)(REGEX)(\\2)\\s*$\\n?'
-            'beginCaptures':
-              '0':
-                'name': 'punctuation.section.embedded.begin.php'
+            'include': '#interpolation'
+          }
+        ]
+      }
+    ]
+  'nowdoc_interior':
+    'patterns': [
+      {
+        'begin': '(<<<)\\s*\'(HTML)\'(\\s+)?$'
+        'beginCaptures':
+          '0':
+            'name': 'punctuation.section.embedded.begin.php'
+          '1':
+            'name': 'punctuation.definition.string.php'
+          '2':
+            'name': 'keyword.operator.heredoc.php'
+          '3':
+            'name': 'invalid.illegal.trailing-whitespace.php'
+        'contentName': 'text.html'
+        'end': '^(\\2)\\b'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.embedded.end.php'
+          '1':
+            'name': 'keyword.operator.heredoc.php'
```

Comment:
(assuming you meant for `nowdoc_interior` and not `heredoc_interior`) yes please.